controllers/Challenges.js

```diff
@@ -7,23 +7,27 @@ const Performance = new models.Performance();
 function ChallengersController() {
   this.create = (req, res) => {
     const { spotId } = req.body;
-    const userId = req.user.nameNumber, performanceId = req.session.currentPerformance.id;
+    const userId = req.user.nameNumber, performanceId = req.session.currentPerformance && req.session.currentPerformance.id;
 
-    logger.challengesLog(`${req.user.name} sent request to challenge ${spotId}`);
-    return Challenge.create(userId, spotId, performanceId)
-      .then((code) => {
-        email.sendChallengeSuccessEmail({
-          email: req.user.email,
-          performanceName: req.session.currentPerformance.name,
-          spotId
+    if (performanceId) {
+      logger.challengesLog(`${req.user.name} sent request to challenge ${spotId}`);
+      return Challenge.create(userId, spotId, performanceId)
+        .then((code) => {
+          email.sendChallengeSuccessEmail({
+            email: req.user.email,
+            performanceName: req.session.currentPerformance.name,
+            spotId
+          });
+          logger.challengesLog(`${req.user.name} successfully challenged for ${spotId}`);
+          res.json({ code });
+        })
+        .catch((err) => {
+          logger.errorLog('Challenges.create', err);
+          res.status(400).send(err);
         });
-        logger.challengesLog(`${req.user.name} successfully challenged for ${spotId}`);
-        res.json({ code });
-      })
-      .catch((err) => {
-        logger.errorLog('Challenges.create', err);
-        res.status(400).send(err);
-      });
+    } else {
+      return res.json({ code: 4 });
```

Comment:
https://github.com/osumb/challenges/blob/master/db/schema.sql#L126-L145 Is where that gets defined. It's kind of sloppy. `code: 4` will just make an arbitrary error message. It definitely needs to be documented better. The db schema is a bad place for it